### ISSUE TITLE
deadCodeInjection doesn't work in webpack-obfuscator

### ISSUE THREAD

**Issue by ninjajomarie** [javascript-obfuscator/javascript-obfuscator#769]
javascript-obfuscator and webpack-obfuscator are amazing.
I love them and I have been using them in my several project.
I happened to find out an issue when I tried to use webpack-obfuscator in a vue-electron application.

Here is my vue.configu.js
var WebpackObfuscator = require('webpack-obfuscator');
module.exports = {
    pluginOptions: {
        electronBuilder: {
            // Use this to change the entrypoint of your app's main process
            mainProcessFile: 'app.js',
        }
    },
    configureWebpack: {
        plugins: [
            new WebpackObfuscator (
                {
                    compact: true,
                    controlFlowFlattening: false,
                    controlFlowFlatteningThreshold: 0.2,
                    deadCodeInjection: true,
                    deadCodeInjectionThreshold: 0.5,
                    debugProtection: false,
                    debugProtectionInterval: false,
                    disableConsoleOutput: false, // change to true
                    identifierNamesGenerator: 'hexadecimal',
                    log: false,
                    numbersToExpressions: false,
                    renameGlobals: false,
                    rotateStringArray: true,
                    selfDefending: true,
                    shuffleStringArray: true,
                    simplify: true,
                    splitStrings: false,
                    // splitStringsChunkLength: 10,
                    stringArray: true,
                    stringArrayEncoding: [],
                    stringArrayWrappersCount: 1,
                    stringArrayWrappersChainedCalls: true,
                    stringArrayWrappersType: 'variable',
                    stringArrayThreshold: 0.75,
                    // transformObjectKeys: true,
                    unicodeEscapeSequence: false
                },
                []
            )
        ]
    }
}

I can build the app successfully with above options but the app has a problem when running.
Here is the error in console. "chunk-vendors.36e4f893.js:11 Uncaught SyntaxError: Unexpected reserved word"
If I set false in deadCodeInjection, the app runs smoothly.
In a nutshell the deadCodeInjection in the webpack-obfuscator plugin doesn't work.
Can you please help me to figure it out?
Thanks in advance.

**Comment by sanex3339** [comment]
Hi. I need additional info.

**Comment by ninjajomarie** [comment]
could you please let me know in detail what you need?

**Comment by sanex3339** [comment]
Obfuscator version, electron version, more details about the error. Obfuscated code if possible. Try to create a reproducible example.

**Comment by ninjajomarie** [comment]
"@vue/cli-plugin-babel": "~4.5.4",
        "@vue/cli-plugin-eslint": "~4.5.4",
        "@vue/cli-service": "~4.5.4",
        "babel-eslint": "^10.1.0",
        "electron": "^10.1.1",
        "eslint": "^7.8.1",
        "eslint-plugin-vue": "^6.2.2",
        "javascript-obfuscator": "^2.4.2",
        "vue-cli-plugin-electron-builder": "^1.4.6",
        "vue-template-compiler": "^2.6.12",
        "webpack-obfuscator": "^2.4.2"


and here is the screenshot of the error
https://prnt.sc/ure4nb

this is the js code obfuscated.
https://we.tl/t-GU5aAf8FuE

**Comment by sanex3339** [comment]
Can you send me the same screenshot of the error and obfuscated code but with `compact: false` option

**Comment by ninjajomarie** [comment]
hold on please

**Comment by ninjajomarie** [comment]
here is the screenshot
https://prnt.sc/urecxp
and js
https://we.tl/t-H39fwSQfU6

**Comment by sanex3339** [comment]
Thank you, will check it

**Comment by ninjajomarie** [comment]
thank you

**Comment by sanex3339** [comment]
The problem with `for await of` construction:
```
for await (const _0x3a4bd3 of _0x6f2186(_0x50d9b9, _0x4d733e))
     _0xf3d53e[_0x477e2('0x1a84')](_0x3a4bd3);
```

**Comment by ninjajomarie** [comment]
could you possibly let me know how I can resolve the issue?

**Comment by sanex3339** [comment]
This is the problem on the obfuscator side. This construction shouldn't be used as dead code

**Comment by ninjajomarie** [comment]
there are piles of sections using "await".
are there any options not to obfuscate them only?

**Comment by sanex3339** [comment]
I'll fix it soon

**Comment by sanex3339** [comment]
PR: https://github.com/javascript-obfuscator/javascript-obfuscator/pull/770

**Comment by sanex3339** [comment]
Merged into `master` branch. Will release it soon.

**Comment by ninjajomarie** [comment]
thank you

**Comment by sanex3339** [comment]
Released as `2.4.3`

**Comment by ninjajomarie** [comment]
awesome!

**Comment by ninjajomarie** [comment]
Excellent!
The issue completely fixed!
